Index: xdm-X11R7.0-1.0.1/auth.c
===================================================================
--- xdm-X11R7.0-1.0.1.orig/auth.c	2006-03-05 20:46:28.000000000 -0500
+++ xdm-X11R7.0-1.0.1/auth.c	2006-03-05 20:47:28.000000000 -0500
@@ -60,9 +60,7 @@
 # include <netdnet/dnetdb.h>
 #endif
 
-#if defined(hpux)
 #include <sys/utsname.h>
-#endif
 
 #if defined(SYSV) && defined(i386)
 # include <sys/stream.h>
