From: Branden Robinson <branden@debian.org>
Subject: [PATCH] Use xdm's _SysErrorMsg() function instead of merely printing errno

---
 greeter/verify.c |    2 +-
 session.c        |    7 ++++---
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/greeter/verify.c b/greeter/verify.c
index fad60c9..e9aeccb 100644
--- a/greeter/verify.c
+++ b/greeter/verify.c
@@ -446,7 +446,7 @@ Verify (struct display *d, struct greet_info *greet, struct verify_info *verify)
 	errno = 0;
 	sp = getspnam(greet->name);
 	if (sp == NULL) {
-	    Debug ("getspnam() failed, errno=%d.  Are you root?\n", errno);
+	    Debug ("getspnam() failed: %s\n", _SysErrorMsg (errno));
 	} else {
 	    user_pass = sp->sp_pwdp;
 	}
diff --git a/session.c b/session.c
index b48a3ef..3c464ab 100644
--- a/session.c
+++ b/session.c
@@ -680,7 +680,8 @@ StartClient (
 	 * audit classes, user limits, and umask.
 	 */
 	if (setpcred(name, NULL) == -1) {
-	    LogError ("setpcred for \"%s\" failed, errno=%d\n", name, errno);
+	    LogError ("setpcred for \"%s\" failed: %s\n", name,
+		      _SysErrorMsg (errno));
 	    return (0);
 	}
 #endif /* AIXV3 */
@@ -804,8 +805,8 @@ StartClient (
 	if (passwd != NULL)
 	    bzero(passwd, strlen(passwd));
 	Debug ("StartSession, fork failed\n");
-	LogError ("can't start session on \"%s\", fork failed, errno=%d\n",
-		  d->name, errno);
+	LogError ("can't start session on \"%s\", fork failed: %s\n",
+		  d->name, _SysErrorMsg (errno));
 	return 0;
     default:
 	if (passwd != NULL)
-- 
1.5.6.3

