From: Branden Robinson <branden@debian.org>
Subject: [PATCH] Use xdm's _SysErrorMsg() function instead of strerror()

---
 choose.c |    2 +-
 daemon.c |    6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/choose.c b/choose.c
index ad96800..1d8a88f 100644
--- a/choose.c
+++ b/choose.c
@@ -502,7 +502,7 @@ ProcessChooserSocket (int fd)
     }
     else
     {
-	LogError ("Choice response read error: %s\n", strerror(errno));
+	LogError ("Choice response read error: %s\n", _SysErrorMsg(errno));
     }
 
 #if defined(STREAMSCONN)
diff --git a/daemon.c b/daemon.c
index 34cbea7..3738b24 100644
--- a/daemon.c
+++ b/daemon.c
@@ -52,14 +52,14 @@ BecomeDaemon (void)
 #ifdef HAVE_DAEMON
     if (daemon (0, 0) < 0) {
        /* error */
-       LogError("daemon() failed, %s\n", strerror(errno));
+       LogError("daemon() failed, %s\n", _SysErrorMsg (errno));
        exit(1);
     }
 #else
     switch (fork()) {
     case -1:
        /* error */
-       LogError("daemon fork failed, %s\n", strerror(errno));
+       LogError("daemon fork failed, %s\n", _SysErrorMsg (errno));
        exit(1);
        break;
     case 0:
@@ -72,7 +72,7 @@ BecomeDaemon (void)
 
     if (setsid() == -1) {
        LogError("setting session id for daemon failed: %s\n",
-                  strerror(errno));
+                  _SysErrorMsg (errno));
        exit(1);
     }
 
-- 
1.5.6.3

