Index: xdm/xdm.man.cpp
===================================================================
--- xdm.orig/xdm.man.cpp
+++ xdm/xdm.man.cpp
@@ -1473,7 +1473,8 @@
 .IR Xserver (__appmansuffix__),
 .\" .IR chooser (__appmansuffix__), \" except that there isn't a manual for it yet
 .\" .IR xdmshell (__appmansuffix__), \" except that there isn't a manual for it yet
-.IR fonts.conf (__filemansuffix__).
+.IR fonts.conf (__filemansuffix__),
+.IR xdm.options (__filemansuffix__).
 .br
 .I "X Display Manager Control Protocol"
 .br
